src/components/Proposal/Voting/VoteButtons.tsx

```diff
@@ -120,60 +124,139 @@ class VoteButtons extends React.Component<IContainerProps, IState> {
-                <img className={css.buttonLoadingImg} src="/assets/images/Icon/buttonLoadingBlue.gif"/>
-                <span> Against</span>
-              </button>
+              <span className={css.hasVoted}>
+                You voted
+              </span>
+              <span className={css.hasNotVoted}>
+                Vote
+              </span>
             </div>
-            :
-            <div className={css.votingDisabled}>
-              <Tooltip overlay={tipContent}>
-                <span>Voting disabled</span>
-              </Tooltip>
+            {votingDisabled ?
+              <span className={css.votingDisabled}>
+                Voting disabled
+              </span>
+              : " "
+            }
+          </div>
+          <div className={css.contextContent}>
+            <div className={css.hasVoted}>
+              <div className={css.voteRecord}>
+                <span className={css.castVoteFor} data-test-id="youVotedFor">
+                 <img src="/assets/images/Icon/Vote/against-btn-fill-red.svg"/>
+                 <br/>
+                 For
+                </span>
+                <span className={css.castVoteAgainst}>
+                 <img src="/assets/images/Icon/Vote/for-fill-green.svg"/>
+                 <br/>
+                 Against
+                </span>
+              </div>
             </div>
-          }
+            <div className={css.hasNotVoted}>
+              {!votingDisabled ?
+                <div>
+                  <button onClick={this.handleClickVote.bind(this, 1)} className={voteUpButtonClass} data-test-id="voteFor">
+                    <img src={`/assets/images/Icon/vote/for-btn-selected${altStyle ? "-w" : ""}.svg`} />
+                    <img className={css.buttonLoadingImg} src="/assets/images/Icon/buttonLoadingBlue.gif"/>
+                    <span> For</span>
+                  </button>
+                  <button onClick={this.handleClickVote.bind(this, 2)} className={voteDownButtonClass}>
+                    <img src={`/assets/images/Icon/vote/against-btn-selected${altStyle ? "-w" : ""}.svg`}/>
+                    <img className={css.buttonLoadingImg} src="/assets/images/Icon/buttonLoadingBlue.gif"/>
+                    <span> Against</span>
+                  </button>
+                </div>
+                :
+                <div className={css.votingDisabled}>
+                  <Tooltip overlay={tipContent}>
+                    <span>Voting disabled</span>
+                  </Tooltip>
+                </div>
+              }
+            </div>
+            {votingDisabled ?
+              <span className={css.votingDisabled}>
+                You do not have enough reputation
+              </span>
+              : " "
+            }
+          </div>
         </div>
+      );
+    } else {
+      return (
+        <div className={wrapperClass} >
+          {this.state.showPreVoteModal ?
```

Comment:
again id like to figure out what doesnt need to be duplicated, like this probably doesnt

Reply:
Happy to look at it together... perhaps they should be two separate components?